client/src/metrics.rs

```diff
@@ -167,13 +205,29 @@ impl Service<http::Request<BoxBody>> for GrpcMetricSvc {
                 })
             });
         let callfut = self.inner.call(req);
+        let errcode_label_disabled = self.disable_errcode_label;
         async move {
             let started = Instant::now();
             let res = callfut.await;
             if let Some(metrics) = metrics {
                 metrics.record_svc_req_latency(started.elapsed());
-                if res.is_err() {
-                    metrics.svc_request_failed();
+                if let Ok(ref ok_res) = res {
+                    if let Some(number) = ok_res
+                        .headers()
+                        .get("grpc-status")
+                        .and_then(|s| s.to_str().ok())
+                        .and_then(|s| s.parse::<i32>().ok())
```

Comment:
Ug, I wonder what the Tonic recommended way is to access the full gRPC status in an interceptor-like pattern or if it's just not supported.

Reply:
It's not really, at least not directly in any simple way.